lib/sequin/runtime/slot_message_store.ex

```diff
@@ -1001,6 +1004,51 @@ defmodule Sequin.Runtime.SlotMessageStore do
     end)
   end
 
+  defp maybe_pull_messages_from_disk(%State{all_loaded?: true} = state), do: state
+
+  defp maybe_pull_messages_from_disk(%State{all_loaded?: false} = state) do
+    should_pull_more_messages? =
+      state.payload_size_bytes < div(state.max_memory_bytes, 2) and
+        map_size(state.messages) < div(state.setting_max_messages, 2)
+
+    if should_pull_more_messages? do
+      # Stream messages and stop when we reach max_memory_bytes
+      {persisted_messages, _current_size_bytes, _message_count, all_loaded?} =
+        state.consumer
+        |> Consumers.stream_consumer_messages_for_consumer()
+        |> Stream.filter(&(message_partition(&1, state.consumer.partition_count) == state.partition))
+        |> Enum.reduce_while({[], 0, 0, true}, fn msg, {messages, current_size, message_count, _all_loaded?} ->
+          if State.message_exists?(state, msg) do
+            {:cont, {messages, current_size, message_count, true}}
+          else
+            msg = %{msg | payload_size_bytes: :erlang.external_size(msg.data)}
+            new_size = current_size + msg.payload_size_bytes
+            new_message_count = message_count + 1
+
+            if new_size <= state.max_memory_bytes and new_message_count <= state.setting_max_messages do
+              {:cont, {[msg | messages], new_size, new_message_count, true}}
+            else
+              Logger.info("[SlotMessageStore] Reached memory limit, not loading more persisted messages",
```

Comment:
a warning level might be more suitable for this message
```suggestion
              Logger.warning("[SlotMessageStore] Reached memory limit, not loading more persisted messages",
```

Reply:
agree 👍